### PR TITLE
usb: fixed composite usbdev issue

### PR DESCRIPTION
**Pull request by zyfeier**
## Summary

1. adb: When the adbd task is running, the usb uninstall process will panic in the adbd task, because the memory of the adb driver has been freed.
2. cdcacm: Logic error handled in USB_REQ_SETINTERFACE.
3. composite: fix enumeration error for 192 bytes desc.
4. composite: The maximum number of composite devices supported is changed to 8.
5. composite: In the device descriptor, 0xEF means use of IAD.

## Impact

USB

## Testing

sim:usbdev

### CODE REVIEW

File: drivers/usbdev/composite.h [hunk 1]
```diff
@@ -95,7 +95,7 @@
 #  define COMPOSITE_REMOTEWAKEUP      (0)
 #endif
 
-#define NUM_DEVICES_TO_HANDLE         (4)
+#define NUM_DEVICES_TO_HANDLE         (8)
```
Comment (raiden00pl) [inline drivers/usbdev/composite.h:98]: shouldn't we make `NUM_DEVICES_TO_HANDLE` configurable with Kconfig ?
then we can reuse this value in the board initialization logic, like here https://github.com/apache/nuttx/blob/856526adee25cdb9987ff800767a71ec9059d1ea/boards/arm/stm32h7/nucleo-h743zi/src/stm32_composite.c#L178 which is now not possible as `NUM_DEVICES_TO_HANDLE` is not available outside `drivers/usbdev`

**Review by raiden00pl — APPROVED** [review]

File: drivers/usbdev/composite.h (hunk 1 above)
Comment (zyfeier) [reply drivers/usbdev/composite.h:98]: Hi, raiden00pl, this is a good idea, i will submit a new PR for this change.

File: drivers/usbdev/adb.c [hunk 2]
```diff
@@ -1625,10 +1628,28 @@ static void usbclass_uninitialize(FAR struct usbdevclass_driver_s *classdev)
...
+#endif
+        }
+
+      return;
+    }
 
   unregister_driver(USBADB_CHARDEV_PATH);
 
-  kmm_free(alloc);
+  if (alloc->dev.registered)
```
Comment (pkarashchenko) [inline drivers/usbdev/adb.c:1645]: Could be simplified to use `esle` and remove `return;` statements

File: drivers/usbdev/rndis.c [hunk 3]
```diff
@@ -2942,15 +2942,21 @@ static void usbclass_uninitialize(FAR struct usbdevclass_driver_s *classdev)
...
   FAR struct rndis_alloc_s *alloc = (FAR struct rndis_alloc_s *)drvr->dev;
-
+  if (!alloc->dev.registered)
+    {
+#ifdef CONFIG_USBADB_COMPOSITE
+      kmm_free(alloc);
+#endif
+      return;
+    }
   if (drvr->dev->registered)
```
Comment (pkarashchenko) [inline drivers/usbdev/rndis.c:2952]: Can do `else if` and remove `return;`

File: drivers/usbdev/rndis.c [hunk 4]
```diff
@@ -2942,15 +2942,21 @@ static void usbclass_uninitialize(FAR struct usbdevclass_driver_s *classdev)
...
     {
       netdev_unregister(&drvr->dev->netdev);
       drvr->dev->registered = false;
-    }
-  else
-    {
+#ifndef CONFIG_RNDIS_COMPOSITE
       kmm_free(alloc);
+#endif
+      return;
```
Comment (pkarashchenko) [inline drivers/usbdev/rndis.c:2959]: `return;` is not needed here

**Pull request merged**